### PR TITLE
Upload page improvements

### PR DESCRIPTION
**Pull request by gbaz**
Some cleanup and modernization of the upload page.

### CODE REVIEW

**Comment by hvr** [comment]
@sclv is this PR still ready to go? if so I'd like to pick it up for next weekend's small-change deployment

File: datafiles/templates/upload.html.st [hunk 1]
```diff
@@ -72,37 +66,32 @@ See the notes at the bottom of the page.
...
-    on making packages work with a range of versions of GHC.
-  </li>
-
-  <li>
-    While <a href="http://www.haskell.org/haddock/">Haddock 2</a>
-    accepts GHC features, it is also more picky about comment syntax than
-    the old version.
-  </li>
+cabal haddock --builddir="$dir" --for-hackage --haddock-option=--hyperlinked-source
+cabal upload -d $dir/*-docs.tar.gz
```
Comment (hvr) [inline datafiles/templates/upload.html.st]: Note that starting with cabal 2.0 one also needs to use the `--publish` flag to `cabal upload` (as it defaults to candidates)

File: datafiles/templates/upload.html.st [hunk 2]
```diff
@@ -72,37 +66,32 @@ See the notes at the bottom of the page.
...
-    See <a href="http://www.haskell.org/haskellwiki/Upgrading_packages">these notes</a>
-    on making packages work with a range of versions of GHC.
-  </li>
-
-  <li>
-    While <a href="http://www.haskell.org/haddock/">Haddock 2</a>
-    accepts GHC features, it is also more picky about comment syntax than
-    the old version.
-  </li>
+cabal haddock --builddir="$dir" --for-hackage --haddock-option=--hyperlinked-source
```
Comment (hvr) [inline datafiles/templates/upload.html.st]: why the `builddir` argument?

**Comment by gbaz** [comment]
@hvr I picked up the script from one you proposed. If you'd like to make any changes to it, go for it!

**Comment by gbaz** [comment]
Other than the changes you'd like to make there, I'd be happy to see this deployed.

**Comment by hvr** [comment]
@gbaz Unfortunately I just realised that this is one of the templates that diverge between the master and central-server branch... which makes this somewhat awkward to merge... :-/

**Pull request merged**